File: C++/575. Distribute Candies.cpp
// Time: O(n)
class Solution {
public:
    int distributeCandies(vector<int>& candies) {
        set<int> ans;
        for(auto candie : candies)
            ans.insert(candie);
        return min(ans.size(), candies.size()/2);
    }
};

File: Python/575. Distribute Candies.py
# Time: O(n)
class Solution:
    def distributeCandies(self, candies):
        """
        :type candies: List[int]
        :rtype: int
        """
        ans = set()
        for candie in candies:
            ans.add(candie)
        return min(len(ans), len(candies)//2)